src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> list;
+        if (type.equals(Bulldozer.class)) {
+            BulldozerProducer bulldozerProducer = new BulldozerProducer();
+            list = bulldozerProducer.get();
+        } else if (Excavator.class.equals(type)) {
+            ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+            list = excavatorProducer.get();
+        } else if (Truck.class.equals(type)) {
+            TruckProducer truckProducer = new TruckProducer();
+            list = truckProducer.get();
+        } else {
+            return new ArrayList<>();
+        }
+        return new ArrayList<>(list);
```

Comment:
The logic is correct, but what do you think, if we just assign smth for list value (for ex.: **list = new SmthProducer()**) in if-statement's body, and after all the checks return list.get() ? It will short your code and you could not create new ArrayList in the end.